### PR TITLE
[IMP] project: improve recurrence notebook and project sharing

### PR DESCRIPTION
**Pull request by bastvdn**
Description of the feature this PR addresses:

- project sharing: display the name of the project in the action instead of 'project sharing'
- project.task: sort records by priority > deadline > sequence > id
- project.task form view > recurrence notebook: indicate the day of the week in the list of date

Task-3034635

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/project/models/project.py [hunk 1]
```diff
@@ -759,6 +760,11 @@ def project_update_all_action(self):
         action['display_name'] = _("%(name)s's Updates", name=self.name)
         return action
 
+    def action_project_sharing(self):
+        action = self.env['ir.actions.act_window']._for_xml_id('project.project_sharing_project_task_action')
+        action['display_name'] = _("%(name)s", name=self.name)
```
Comment (auon-odoo) [inline addons/project/models/project.py:765]: In methods above, `_("%(name)s blabla", name=self.name)` is used to translate 'blabla'.
As there is no 'blabla' to translate here :
```suggestion
        action['display_name'] = self.name
```

File: addons/project/models/project.py [hunk 2]
```diff
@@ -1516,7 +1522,7 @@ def _compute_recurrence_message(self):
             else:
                 recurrence_title = _('A new task will be created on the following dates:')
             task.recurrence_message = '<p><span class="fa fa-check-circle"></span> %s</p><ul>' % recurrence_title
-            task.recurrence_message += ''.join(['<li>%s</li>' % date.strftime(date_format) for date in recurring_dates[:5]])
+            task.recurrence_message += ''.join(['<li>%s %s</li>' % (format_date(date, "EEE", locale=self.env.user.lang), date.strftime(date_format)) for date in recurring_dates[:5]])
```
Comment (auon-odoo) [inline addons/project/models/project.py:1525]:
```suggestion
            task.recurrence_message += ''.join(['<li>%s %s</li>' % (
                format_date(date, "EEE", locale=self.env.user.lang), date.strftime(date_format)
            ) for date in recurring_dates[:5]])
```
...to see it a little clearer

File: addons/project/static/src/project_sharing/project_sharing.js [hunk 3]
```diff
@@ -33,9 +33,9 @@ export class ProjectSharingWebClient extends Component {
     }
 
     async _showView() {
-        const { action_name, project_id, open_task_action } = session;
+        const { action, project_id, open_task_action } = session;
```
Comment (auon-odoo) [inline addons/project/static/src/project_sharing/project_sharing.js:36]: Might be more correct, but logic wise, it is an unnecessary change.
We avoid making those, because they may mess forward-ports up. :(

**Review by auon-odoo — CHANGES_REQUESTED** [review]
Thank you for your work :fire: Left some comments.

File: addons/project/__manifest__.py [hunk 4]
```diff
@@ -113,6 +113,7 @@
             'web/static/src/views/view_button/**/*',
             'web/static/src/views/view_dialogs/**/*',
             'web/static/src/views/widgets/**/*',
+            'web/static/src/views/graph/colors.js',
```
[the remark this answers is not in the document]
Comment (bastvdn) [reply addons/project/__manifest__.py]: Had a bug for the dependencies needed for project sharing,
works in current master branch thanks to one of XBO fix :)

File: addons/project/static/src/project_sharing/project_sharing.js (hunk 3 above)
Comment (bastvdn) [reply addons/project/static/src/project_sharing/project_sharing.js]: yes makes sense sorry :'(

File: addons/project/models/project.py (hunk 1 above)
Comment (bastvdn) [reply addons/project/models/project.py]: Yep, makes sense thanks

File: addons/project/controllers/portal.py [hunk 5]
```diff
@@ -144,10 +144,11 @@ def _prepare_project_sharing_session_info(self, project, task=None):
             "translations": translation_hash,
         }
 
+        action_mod = project.action_project_sharing()
         project_company = project.company_id
         session_info.update(
             cache_hashes=cache_hashes,
-            action_name='project.project_sharing_project_task_action',
+            action_name=action_mod,
```
Comment (auon-odoo) [inline addons/project/controllers/portal.py:151]:
```suggestion
            action_name=project.action_project_sharing(),
```
No need for a variable

File: addons/project/models/project.py [hunk 6]
```diff
@@ -1509,7 +1515,9 @@ def _compute_recurrence_message(self):
             else:
                 recurrence_title = _('A new task will be created on the following dates:')
             task.recurrence_message = '<p><span class="fa fa-check-circle"></span> %s</p><ul>' % recurrence_title
-            task.recurrence_message += ''.join(['<li>%s</li>' % date.strftime(date_format) for date in recurring_dates[:5]])
+            task.recurrence_message += ''.join(['<li>%s %s</li>' % (
+                format_date(date, "EEE", locale=self.env.user.lang), date.strftime(date_format)
+                ) for date in recurring_dates[:5]])
```
Comment (auon-odoo) [inline addons/project/models/project.py:1520]:
```suggestion
            task.recurrence_message += ''.join(['<li>%s %s</li>' % (
                format_date(date, "EEE", locale=self.env.user.lang), date.strftime(date_format)
            ) for date in recurring_dates[:5]])
```
:see_no_evil:

**Review by auon-odoo — CHANGES_REQUESTED** [review]
Last details and we should be good to go :)

**Comment by bastvdn** [comment]
Done :^)
(Is it okay to use the "commit suggestion" option of github for the last small changes you requested, are the commits going to be squashed automatically?)

**Review by auon-odoo — APPROVED** [review]
Thank you :)

File: addons/project/models/project.py [hunk 7]
```diff
@@ -1502,7 +1508,9 @@ def _compute_recurrence_message(self):
             else:
                 recurrence_title = _('A new task will be created on the following dates:')
             task.recurrence_message = '<p><span class="fa fa-check-circle"></span> %s</p><ul>' % recurrence_title
-            task.recurrence_message += ''.join(['<li>%s</li>' % date.strftime(date_format) for date in recurring_dates[:5]])
+            task.recurrence_message += ''.join(['<li>%s %s</li>' % (
+                format_date(date, "EEE", locale=self.env.user.lang), date.strftime(date_format)
+            ) for date in recurring_dates[:5]])
```
Comment (auon-odoo) [inline addons/project/models/project.py:1513]:
```suggestion
            date_format = f"%a {self.env['res.lang']._lang_get(self.env.user.lang).date_format}"
            ...
            task.recurrence_message += ''.join(['<li>%s</li>' % date.strftime(date_format))
            for date in recurring_dates[:5]])
```
Just learned it from Vincent. https://www.programiz.com/python-programming/datetime/strftime
`%a` = Abbreviated weekday name (Sun, Mon, ..)

**Review by auon-odoo — CHANGES_REQUESTED** [review]
Would be cool to change that :)

File: addons/project/models/project.py (hunk 7 above)
Comment (bastvdn) [reply addons/project/models/project.py:1513]: If we want the days of the week to be in the language of the user, we would need to change the 'locale' of the program according to the user language, it seems like a bad practice, Vincent said it'll be more convenient to use babel, can you confirm ?

**Review by auon-odoo — APPROVED** [review]

File: addons/project/models/project.py (hunk 7 above)
Comment (bastvdn) [reply addons/project/models/project.py:1513]: No probs :p thanks for the review

File: addons/project/models/project.py [hunk 8]
```diff
@@ -761,6 +762,11 @@ def project_update_all_action(self):
         action['display_name'] = _("%(name)s's Updates", name=self.name)
         return action
 
+    def action_project_sharing(self):
```
Comment (MissingNoShiny) [inline addons/project/models/project.py:765]: We could add an ensure_one here
```suggestion
    def action_project_sharing(self):
        self.ensure_one()
```

File: addons/project/models/project.py (hunk 7 above)
Comment (xavierbol) [reply addons/project/models/project.py:1514]: I am wondering if we cannot use the following function instead:
https://github.com/odoo/odoo/blob/87307a9010c6c872fe6c746d5ece378ad90d038f/odoo/tools/misc.py#L1359-L1392

What do you think? 🙂

File: addons/project/__manifest__.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (xavierbol) [reply addons/project/__manifest__.py]: Normally, the right fix is the following:

```suggestion
            ('remove', 'web/static/src/views/fields/journal_dashboard_graph/**/*'),  # only works with graph view in assets
```

And normally this fix should come with a PR from the Framework JS but it seems the PR is not yet merged. 🤔 

I will check tomorrow where is that PR to know if we apply the fix in your PR or not. 🤔 🙂

**Review by xavierbol — COMMENTED** [review]
Thanks for your work, I left 2 comments and after that it will look good to me. 🙂 

You can also squash your commits to have only one. 🙂

File: addons/project/__manifest__.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (xavierbol) [reply addons/project/__manifest__.py]: You can ignore my comment. 🙂

File: addons/project/models/project.py [hunk 9]
```diff
@@ -1502,7 +1508,9 @@ def _compute_recurrence_message(self):
             else:
                 recurrence_title = _('A new task will be created on the following dates:')
             task.recurrence_message = '<p><span class="fa fa-check-circle"></span> %s</p><ul>' % recurrence_title
-            task.recurrence_message += ''.join(['<li>%s</li>' % date.strftime(date_format) for date in recurring_dates[:5]])
+            task.recurrence_message += ''.join(['<li>%s %s</li>' % (
+                format_date(self.env, date, date_format="EEE"), date.strftime(date_format)
```
Comment (xavierbol) [inline addons/project/models/project.py:1512]: wonderful. 😁

**Review by xavierbol — APPROVED** [review]
LGTM, thanks for your work. 🙂 

robodoo r+
